src/components/Modal/__tests__/modal-tests.jest.js

```diff
@@ -0,0 +1,175 @@
+import React from "react";
+import { render, cleanup, fireEvent } from "@testing-library/react";
+import Modal from "../Modal";
+import { useModalHelper } from "components/Modal/__stories__/helpers";
+import { ModalHeader } from "components";
+import "@testing-library/jest-dom";
+
+const MODAL_TITLE_TEXT = "Modal title";
+const MODAL_ID = "story-book-modal";
+const CLOSE_BUTTON_LABEL = "close-button";
+const OPEN_BUTTON_TEXT = "Open";
+
+const ModalManager = props => {
+  const { children, openOnStart = false, isAlertDialog = false, title } = props;
+  const { openModalButton, modalProps } = useModalHelper({
+    openButtonTitle: OPEN_BUTTON_TEXT,
+    closeButtonAriaLabel: CLOSE_BUTTON_LABEL,
+    openOnStart
+  });
+  return (
+    <div>
+      {openModalButton}
+      <Modal {...modalProps} id={MODAL_ID} title={title || MODAL_TITLE_TEXT} isAlertDialog={isAlertDialog}>
+        {children}
+      </Modal>
+    </div>
+  );
+};
+
+const getOpenButton = component => component.getByText(OPEN_BUTTON_TEXT);
+const findModal = component => component.findByRole("dialog");
+const queryModal = component => component.queryByRole("dialog");
+const queryClosedModal = component => component.queryByTestId("monday-dialog-container");
+const queryAlertModal = component => component.queryByRole("alertdialog");
+const findCloseButton = component => component.findByLabelText(CLOSE_BUTTON_LABEL);
+const queryCloseButton = component => component.queryByLabelText(CLOSE_BUTTON_LABEL);
+const findOverlay = component => component.findByTestId("monday-modal-overlay");
+
+const renderComponent = (props = {}) => {
+  const { content } = props;
+  return render(<ModalManager {...props}>{content ? content : <div>content of the modal</div>}</ModalManager>);
+};
+
+describe("Modal tests", () => {
+  afterEach(() => {
+    cleanup();
+  });
+
+  describe("open", () => {
+    it("should open when triggered", async () => {
+      const component = renderComponent({ openOnStart: false });
```

Comment:
Maybe it is just me but it,s weird to me that we test the modal component with so many wrapper. this way things can work or not work only because of one of the 2 wrappers logic or defaults. For example I think it would more reliable to test if pass show to the modal component trigger showing the modal instead of if openOnStart trigger showing the modal. Because in this case if other developer create a bug he can fix the ModalManager/hook which both are not exported outside instead of the component.